documentation/design/proposals/kafka-address-space.adoc

```diff
@@ -0,0 +1,177 @@
+== Overview
+
+This document describes the design of the `kafka` address space type and how it integrates with
+EnMasse.
+
+== Design
+
+
+A tenant creates a kafka address space by specifying the `kafka` address space type when creating an
+`AddressSpace`. 
+
+A service admin creates `KafkaConfig` resources describing the Kafka cluster setup, and
+`AddressSpacePlan` resources referencing those resources.
+
+An address space plan referencing a `KafkaConfig` will look like this:
+
+[source,yaml,options="nowrap",subs="attributes"]
+----
+apiVersion: admin.enmasse.io/v1beta2
+kind: AddressSpacePlan
+metadata:
+  name: kafka-small
+spec:
+  infraConfigRef: kafka-config
+  addressSpaceType: kafka
+  resourceLimits:
+    kafka: 1.0
+----
+
+The `KafkaConfig` resource is a configuration template just like `StandardInfraConfig` and `BrokeredInfraConfig`. The `KafkaConfig` resource uses the schema of the `Kafka` resource in Strimzi and may look like the following:
+
+[source,yaml,options="nowrap",subs="attributes"]
+----
+apiVersion: admin.enmasse.io/v1beta1
+kind: KafkaConfig
+metadata:
+  name: kafka-config
+spec:
+  version: 0.28.0
+  template:
+    kafka:
+      version: 2.1.0
+      replicas: 1
+      listeners:
+        plain: {}
+        tls: {}
+      config:
+        offsets.topic.replication.factor: 1
+        transaction.state.log.replication.factor: 1
+        transaction.state.log.min.isr: 1
+        log.message.format.version: "2.1"
+      storage:
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+    zookeeper:
+      replicas: 1
+      storage:
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+    entityOperator:
+      topicOperator: {}
+      userOperator: {}
+----
+
+Creating this resource will not create a Kafka cluster, but it will prepare the `address-space-controller` with the configuration needed to create one when a tenant creates an `AddressSpace` of the `kafka` type. Moreover, settings that are applied on a per-address-space basis will be ignored and set based on the `AddressSpace` definition instead. The following settings are ignored/overridden:
+
```

Comment:
This mean that all Kafka address spaces share the same Kakfa cluster, right?

Reply:
No, that will come at a later point. I will try to clarify this in the doc, but in the same way we don't create the router and broker when you create the *InfraConfig, we would not create the Kafka cluster when you create the KafkaConfig. The cluster will be created when the AddressSpace is created.

I think longer term, sharing a cluster would be ideal, but it requires modifications to Kafka itself or some proxy in the middle to make it transparent.